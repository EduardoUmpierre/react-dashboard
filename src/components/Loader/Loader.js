import React from 'react'

const loader = ({ isLoading, children }) => {
    let component = <div>Loading</div>

    if (!isLoading) {
        component = children
    }

    return component
}

export default loader
