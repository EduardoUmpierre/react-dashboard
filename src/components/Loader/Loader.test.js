import React from 'react'
import { shallow } from 'enzyme'
import Loader from './Loader'

describe('<Loader />', () => {
    it('renders correctly', () => {
        const component = shallow(<Loader loaded={true}>Test</Loader>)

        expect(component).toMatchSnapshot()
    })

    it('is loading', () => {
        const component = shallow(<Loader loaded={false}>Test</Loader>)

        expect(component.exists('.loader')).toBe(true)
    })

    it('is loaded', () => {
        const component = shallow(<Loader loaded={true}>Test</Loader>)

        expect(component.exists('.loader')).toBe(false)
    })
})
