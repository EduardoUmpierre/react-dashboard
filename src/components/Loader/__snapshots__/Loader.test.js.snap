// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <loader
    loaded={true}
  >
    Test
  </loader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Test",
  Symbol(enzyme.__nodes__): Array [
    "Test",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
